### DISCUSSION TITLE
use ttypescript instead of patching ts-node

### DISCUSSION THREAD

**Discussion by ZPiDER666** [marcj/deepkit#457] [Ideas]
deepkit does some magic in patching ts-node, so it executes the deepkit transformer.
with ts-node updates these patches need to be reapplied and they may even break with some ts-node changes.
i think i managed to do a setup with ttypescript that still executes the ttypescript transformer and it feels cleaner to me, but it currently does not fully work (debugger issue, see issue 445). making this a user choice may not be feasible (unless there would be a separate deepkit package to include with this "flavor").

i would prefer, if ttypescript would be used, as i want to use transformers for other things as well in my projects.

what are your thoughts on this?

**Comment by marcus-sa** [comment]
Deepkit doesn't patch `ts-node`.
It patches `typescript` by adding very little code to an internal function called `getTransformers`, which doesn't interfere with `ttypescript` or `ts-node`.


You can see what it patches in this file
https://github.com/deepkit/deepkit-framework/blob/727afc440d65a44fd8637bebfdb877131b4be51e/packages/type-compiler/install-transformer.ts

**Comment by marcj** [comment]
ttypescript is supported https://github.com/deepkit/deepkit-framework/issues/401#issuecomment-1538280663
note however that ttypescript seems to be unmaintained and doesn't support TS 5+. funny enough, a ttypescript replacement called ts-patch works like what we do currently with patching typescript itself. We had never problems with our mechanism, even with the big TS v5 change (ts-patch didnt work for v5 https://github.com/nonara/ts-patch/issues/93)
